packages/js/product-editor/src/components/block-editor/block-editor.tsx

```diff
@@ -154,6 +159,11 @@ export function BlockEditor( {
 		{ postType }
 	);
 
+	const [ layoutTemplateId ] = useProductEntityProp< string >(
+		'meta_data._layout_template_id',
+		{ postType }
+	);
```

Comment:
I do agree with @louwie17 about having a product template for that because the product template dictates what are the default values to create a new product base on it (those values can be more than just the ones shown as form fields like the product type for example). We do not have that by just having access to the layout itself, so there is no warrantee of the kind of product you want to create based only on the layout.

On the other hand I kind of understand this approach. I know we have not covered this use case yet for the product-variation, it will required an improvement to the system to manage this kind of products. But we need to do it somehow if we want to support more products of any kind in the future.

We should have a way to hide any product template from the product types list, but the idea of having the ability to create new product templates is in fact to let users create products from them, so templates have to be visible somehow. In this case an user cannot create a product like that directly so I understand why we should hide the template from the list.

Reply:
So what are you suggesting @mdperez86 ? stick with this? or add a hide/show functionality?

Btw if this is variation specific, maybe this should be `_product_variation_template_id`?
I also wonder if there is use cases for being able to choose unique variation templates, between different variations? ( There may not be, but its a thought ).

Reply:
I'm in favor to create a product template for that and hide it from the list when need it. What if a specific set of metas are required to create those variations but those metas are not part of the layout template? Keep things homogeneous. Those are the main points why.

Reply:
I had a PR up that took care of this... we decided at the time to shelve it due to larger discussions about templates and such...

https://github.com/woocommerce/woocommerce/pull/43981